pkg/platform/openstack/kuryr_bootstrap.go

```diff
@@ -447,12 +449,40 @@ func waitForOpenStackLb(client *gophercloud.ServiceClient, lbId string) error {
 
 // Looks for a Octavia load balancer by name, address and subnet ID. If it does
 // not exist creates it. Will fail if multiple LB's are matching all criteria.
-func ensureOpenStackLb(client *gophercloud.ServiceClient, name, vipAddress, vipSubnetId string) (string, error) {
-	page, err := loadbalancers.List(client, loadbalancers.ListOpts{
+func ensureOpenStackLb(client *gophercloud.ServiceClient, name, vipAddress, vipSubnetId, tag string) (string, error) {
+	// We need to figure out if Octavia supports tags and use description field if it's too old. To do that
+	// we list available API versions and look for 2.5.
+	allPages, err := apiversions.List(client).AllPages()
+	if err != nil {
+		return "", err
+	}
+
+	apiVersions, err := apiversions.ExtractAPIVersions(allPages)
+	if err != nil {
+		return "", err
+	}
+
+	var octaviaTagSupport bool
+	octaviaTagSupport = false
+	for _, apiVersion := range apiVersions {
+		if apiVersion.ID >= MinOctaviaVersionWithTagSupport {
+			octaviaTagSupport = true
+			break
+		}
+	}
+
+	opts := loadbalancers.ListOpts{
 		Name:        name,
 		VipAddress:  vipAddress,
 		VipSubnetID: vipSubnetId,
-	}).AllPages()
+	}
+	if octaviaTagSupport {
+		opts.Tags = []string{tag}
+	} else {
```

Comment:
Not knowing anything about OpenStack, does it make sense to set both the Tags and Description in the older API case?

Reply:
Not really, the code corresponding to this is here, it simply finds all the LB's having the cluster ID tag in either description or tags: https://github.com/openshift/installer/blob/63c63a0015f8f7cd894a93c0a6c0d6565ee8d05a/pkg/destroy/openstack/openstack.go#L606

Our goal is to support OpenShift's installation on OpenStack back to Queens release. The description is only a workaround that we use to support Queens and Rocky. Once support for those is deprecated we can remove this part. I guess this also needs a comment explaining it.